### PR TITLE
python3Packages.domeneshop: init at 0.4.2

### PR DESCRIPTION
**Pull request by pbsds**
###### Description of changes

Python bindings for the [domene.shop](https://domene.shop/) REST [API](https://api.domeneshop.no/docs/)

###### Things done

I currently use this package in my nixos config to perform dyndns.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/domeneshop/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    inherit pname version;
+    sha256 = "tr3wsrscIU66kTz3nlvDHj2EXoEHCH3grD0yD7BU3Fc=";
+  };
+
+  propagatedBuildInputs = [
+    urllib3
+    pyopenssl
+    cryptography
+    idna
+    certifi
```
Comment (fabaff) [inline pkgs/development/python-modules/domeneshop/default.nix:29]: Use `optional-dependencies` with [`urllib3`](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/python-modules/urllib3/default.nix#L69).

File: pkgs/development/python-modules/domeneshop/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  ];
+
+  # There are none
+  doCheck = false;
+
+  pythonImportsCheck = [ "domeneshop" ];
+
+  meta = with lib; {
+    description = "Python 3 library for working with the Domeneshop API";
```
Comment (fabaff) [inline pkgs/development/python-modules/domeneshop/default.nix:38]: Opinionated change as Python 2 is EOL.

```suggestion
    description = "Python library for working with the Domeneshop API";
```

**Comment by onny** [comment]
Please rebase on top of master, there seems to be some merge conflicts

**Comment by onny** [comment]
Result of `nixpkgs-review pr 185494` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.domeneshop</li>
    <li>python39Packages.domeneshop</li>
  </ul>
</details>

**Pull request merged**